tests/ui/parallel-rustc/unexpected-type-issue-120601.stderr

```diff
@@ -26,13 +23,11 @@ LL | async fn match_() {
    = note: for more on editions, read https://doc.rust-lang.org/edition-guide
 
 error[E0425]: cannot find function, tuple struct or tuple variant `Unstable2` in this scope
-  --> $DIR/unexpected-type-issue-120601.rs:10:5
    |
 LL |     Unstable2(())
    |     ^^^^^^^^^ not found in this scope
 
 error[E0308]: mismatched types
-  --> $DIR/unexpected-type-issue-120601.rs:22:9
```

Comment:
Remark: wait hang on, I knew this diff was fishy. What happened to the file paths or line numbers? I don't think it's with this PR, there's a diagnostics emitter bug or sth here, or maybe with compiletest's rendering of the diagnostics somehow?

EDIT: oh wait, this is compare output by lines. Still feels a bit surprising 🤔

Reply:
I just blessed stuff 🤣. I can rebase the PR on top of a newer commit if that would help.